### PR TITLE
feat(hx-file-tile): surf-1341 readonly file tile

### PR DESCRIPTION
**Pull request by 100stacks**
Update `<hx-file-tile>` to have a `readonly` attribute.

**JIRA: SURF-1341**

### LGTM's
- [x] Dev LGTM
- [x] Design LGTM
- [x] Zoom LGTM

### CODE REVIEW

**Comment by HelixUI** [comment]
Deploy preview for *helix-ui* processing.

Built with commit 94d0521f6b6c9e898db27247a20578c4afb587b8

https://app.netlify.com/sites/helix-ui/deploys/5bb28e6c4ed62f4fdb03d8c3

**Comment by 100stacks** [comment]
Noticed this in `HXFileTileElement.less`: https://github.com/rackerlabs/helix-ui/blob/master/src/helix-ui/elements/HXFileTileElement.less#L29-L33

```css
hx-icon {
      // FIXME: fix in SURF-1219
      margin-left: 0.25rem !important;
    }
  }
```

`SURF-1219` is closed, so not sure if this comment is still relevant.

**Comment by 100stacks** [comment]
@araiford @palindromebart: please review in reference to `SURF-1341`.

kapture_file-tile_readonly

**Comment by 100stacks** [comment]
Docs update:

hx file tile helixui

File: src/helix-ui/elements/HXFileTileElement.js [hunk 1]
```diff
@@ -154,7 +174,7 @@ export class HXFileTileElement extends HXElement {
      * @type {String}
      */
     get name () {
-        return this.getAttribute('name' || '');
+        return this.getAttribute('name') || '';
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileTileElement.js]: Seems we need to do some rebasing of branches. This was merged to `master`, but doesn't seem to be present in `css-audits` or any of its child branches.

**Comment by araiford** [comment]
:shipit: LGTM

File: docs/elements/hx-file-tile/index.html [hunk 2]
```diff
@@ -102,6 +102,13 @@ <h2 id="methods">Methods</h2>
         When present, marks element as loading and denotes completion percentage.
       </p>
     </dd>
+
+    <dt>readonly {Boolean} <i>(optional)</i></dt>
+    <dd>
+      <p>
+        When present, user will not be able to dismiss/delete the file.
```
Comment (CITguy) [inline docs/elements/hx-file-tile/index.html]:
```
When present, marks element as readonly.
```

File: docs/components/files/file-tile-demo.js [hunk 3]
```diff
@@ -41,6 +44,18 @@
                 isInvalid: function () {
                     return this.state === 'invalid' ;
                 },
+                snippet: function () {
```
Comment (CITguy) [inline docs/components/files/file-tile-demo.js]: Rename to `downloadableSnippet`.

File: docs/components/files/index.html [hunk 4]
```diff
@@ -174,20 +181,14 @@ <h2 id="file-tile">File Tile</h2>
...
         <footer>
-          {% code 'html' %}{% raw %}
-            <hx-file-tile
-              details="{{details}}"
-              href="{{href}}"
-              icon="{{icon}}"
-              name="{{name}}">
-            </hx-file-tile>{% endraw %}
-          {% endcode %}
+          <pre><code v-text="snippet"></code></pre>
```
Comment (CITguy) [inline docs/components/files/index.html]: `snippet` -> `downloadableSnippet`

File: docs/components/files/file-tile-demo.js [hunk 5]
```diff
@@ -21,6 +23,7 @@
                 href: 'path/to/id_rsa.pub',
                 icon: ICONS[1],
                 icons: ICONS,
+                readonly: false,
```
Comment (CITguy) [inline docs/components/files/file-tile-demo.js]: alphabetize, please

File: docs/components/files/index.html [hunk 6]
```diff
@@ -159,6 +159,13 @@ <h2 id="file-tile">File Tile</h2>
               <label for="txtDetails">Details</label>
               <input id="txtDetails" class="hxTextCtrl" v-model="details" />
             </p>
+            <fieldset>
+              <legend>Options</legend>
+              <label>
+                <input type="checkbox" v-model="readonly">
```
Comment (CITguy) [inline docs/components/files/index.html]: please self-close tag `<input ... />`

File: docs/components/files/index.html [hunk 7]
```diff
@@ -159,6 +159,13 @@ <h2 id="file-tile">File Tile</h2>
...
+                Read Only
```
Comment (CITguy) [inline docs/components/files/index.html]: One word, please.

File: docs/components/files/test-file-tiles.html [hunk 8]
```diff
@@ -131,6 +131,18 @@ <h3>Downloadable with Custom Name</h3>
...
+        <hx-file-tile
+          details="50B"
+          href="files/easteregg.txt"
+          icon="mime-text"
+          name="downloadable-easter-egg.txt"
+          readonly
+        ></hx-file-tile>
+        <small>readonly download</small>
+      </div>
+
```
Comment (CITguy) [inline docs/components/files/test-file-tiles.html]: Need additional tests:

- [ ] visually truncated + readonly (make sure name can extend all the way to the right edge)
- [ ] loading + readonly
- [ ] error + visually truncated + readonly

File: docs/elements/hx-file-tile/index.html [hunk 9]
```diff
@@ -102,6 +102,13 @@ <h2 id="methods">Methods</h2>
...
+      </p>
+    </dd>
   </dl>
 {% endblock %}
```
Comment (CITguy) [inline docs/elements/hx-file-tile/index.html]: Add documentation for the `readonly` property, as well.

File: src/helix-ui/elements/HXFileTileElement.js [hunk 10]
```diff
@@ -76,6 +77,7 @@ export class HXFileTileElement extends HXElement {
             'icon',
             'name',
             'progress',
+            'readonly',
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileTileElement.js]: please remove this from observed attributes

File: src/helix-ui/elements/HXFileTileElement.js [hunk 11]
```diff
@@ -113,6 +115,24 @@ export class HXFileTileElement extends HXElement {
         return re.exec(this.name)[1] || '';
     }
 
+    /**
+     * If present, the HXFileTileElement is read only and the dismiss will not
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileTileElement.js]:
```
If present, the dismiss button will not be shown.
```

File: src/helix-ui/elements/HXFileTileElement.less [hunk 12]
```diff
@@ -180,3 +180,9 @@
     display: block;
   }
 }
+
+:host([readonly]) {
+  #hxDismiss {
+    display: none;
+  }
```
Comment (CITguy) [inline src/helix-ui/elements/HXFileTileElement.less]: Make sure to apply padding all the way around the `#hxContentWrapper`, when readonly.

File: docs/components/files/test-file-tiles.html (hunk 8 above)
Comment (100stacks) [reply docs/components/files/test-file-tiles.html]: @CITguy, I'm also adding an additional test to match the others:

- [ ] loading + visually truncated + readonly

File: docs/elements/hx-file-tile/index.html (hunk 9 above)
Comment (100stacks) [reply docs/elements/hx-file-tile/index.html]: Additional documentation for the `readonly` property:

        With the `readonly` attribute present, the element will not be dismissible.

File: src/helix-ui/elements/HXFileTileElement.less (hunk 12 above)
Comment (100stacks) [reply src/helix-ui/elements/HXFileTileElement.less]: Padding added to `#hxContentWrapper` to make it `0.5rem` all around, when readonly:

```css
:host([readonly]) {
  #hxDismiss {
    display: none;
  }

  #hxContentWrapper {
    padding: 0.5rem;
  }
}
```

**Comment by 100stacks** [comment]
> Noticed this in `HXFileTileElement.less`: https://github.com/rackerlabs/helix-ui/blob/master/src/helix-ui/elements/HXFileTileElement.less#L29-L33
> 
> ```css
> hx-icon {
>       // FIXME: fix in SURF-1219
>       margin-left: 0.25rem !important;
>     }
>   }
> ```
> 
> `SURF-1219` is closed, so not sure if this comment is still relevant.

Based on @CITguy feedback, removing `SURF-1219` from the comment.  We'll still need to address at a later date.

File: docs/elements/hx-file-tile/index.html [hunk 13]
```diff
@@ -158,6 +165,14 @@ <h2 id="methods">Methods</h2>
       </p>
     </dd>
 
+    <dt>readonly {Boolean} <i>(optional)</i></dt>
+    <dd>
+      <p>
+        With the <code>readonly</code> attribute present, the element will not
```
Comment (CITguy) [inline docs/elements/hx-file-tile/index.html]:
```
Manipulates the <code>readonly</code> attribute.
```

**Pull request merged**